app/controllers/pieces_controller.rb

```diff
@@ -2,6 +2,7 @@ class PiecesController < ApplicationController
   before_action :authenticate_user!
   before_action :player_one_can_only_move_white_and_player_two_can_only_move_black, only: [:show, :update]
```

Comment:
what if we added :castle to the actions that `:player_one_can_only_move_white_and_player_two_can_only_move_black` is called on before they trigger? 
`[:show, :update, :castle]`

Reply:
I tried that but since `pieces#castle` uses the :piece_id and :rook_id to differentiate between the two ids and `:player_one_can_only_move_white_and_player_two_can_only_move_black` uses just :id they did not work together.  I played around with it a bit but could not really get it to work. This is definitely a work around and almost certainly not the best way to do it. I do think that it will work thought. But yes definitely open to suggestions.